packages/app/src/cli/services/extensions/bundle.test.ts

```diff
@@ -0,0 +1,213 @@
+import {bundleExtension} from './bundle.js'
+import {testApp, testUIExtension} from '../../models/app/app.test-data.js'
+import {describe, expect, test, vi} from 'vitest'
+import {build as esBuild, BuildOptions, WatchMode} from 'esbuild'
+import {abort, path} from '@shopify/cli-kit'
+
+vi.mock('esbuild', async () => {
+  const esbuild: any = await vi.importActual('esbuild')
+  return {
+    ...esbuild,
+    build: vi.fn(),
+  }
+})
+
+describe('buildExtension', () => {
+  test('invokes ESBuild with the right options and forwards the logs', async () => {
```

Comment:
Little nitpick, I think we usually use `it` not `test`.

Reply:
I agree we should be more consistent here. I'll place a linting rule and do some renames in a follow-up PR.